tgui/packages/tgui/interfaces/Uplink.js

```diff
@@ -149,39 +170,222 @@ const ItemsPage = (_properties, context) => {
-                      item: i.obj_path,
-                    })
-                  }
-                  disabled={i.cost > crystals}
-                />
-              }
-            >
-              <Box italic>{decodeHtmlEntities(i.desc)}</Box>
-            </Section>
+            <UplinkItem i={i} showDecription={showDesc} key={decodeHtmlEntities(i.name)}/>
           ))}
         </Flex.Item>
       </Flex>
     </Section>
   );
 };
 
+const CartPage = (_properties, context) => {
+  const { act, data } = useBackend(context);
+  const { cart, crystals, cart_price } = data;
+
+  const [showDesc, setShowDesc] = useLocalState(context, "showDesc", 0);
+
+  return (
+    <Fragment>
+      <Section
+        title={'Current Balance: ' + crystals + 'TC'}
+        buttons={
+          <Fragment>
+            <Button.Checkbox
+              content="Show Descriptions"
+              checked={showDesc}
+              onClick={() => setShowDesc(!showDesc)}
+            />
+            <Button
+              content="Empty Cart"
+              icon="trash"
+              onClick={() => act('empty_cart')}
+              disabled={!cart}
+            />
+            <Button
+              content={"Purchase Cart (" + cart_price + "TC)"}
+              icon='shopping-cart'
+              onClick={() => act('purchase_cart')}
+              disabled={!cart || cart_price > crystals}
+            />
+          </Fragment>
+        }
+      >
+        <Flex.Item grow={1} basis={0}>
+          {cart ? cart.map((i) => (
+            <UplinkItem
+              i={i}
+              showDecription={showDesc}
+              key={decodeHtmlEntities(i.name)}
+              buttons={<CartButtons i={i}/>}
+            />
+          ))
+          :
+          <Box italic>Your Shopping Cart is empty!</Box>
+        }
+        </Flex.Item>
+      </Section>
+      <Advert/>
+    </Fragment>
+  )
+}
+const Advert = (_properties, context) => {
+  const { act, data } = useBackend(context);
+  const { cats, lucky_numbers } = data;
+
+  return (
+    <Section
+      title="Suggested Purchases"
+      buttons={
+        <Button
+          icon='dice'
+          content='See more suggestions'
+          onClick={() => act('shuffle_lucky_numbers')}
+        />
+      }
+    >
+      <Flex grow={1} basis={0} mb="5px">
+        <Flex.Item px="5px" width="50%">
+          <UplinkItem i={cats[lucky_numbers[0].cat].items[lucky_numbers[0].item]}/>
+        </Flex.Item>
+        <Flex.Item px="5px" width="50%">
+          <UplinkItem i={cats[lucky_numbers[1].cat].items[lucky_numbers[1].item]}/>
+        </Flex.Item>
+      </Flex>
+      {/* <br/> */}
+      <Flex grow={1} basis={0}>
+        <Flex.Item px="5px" width="50%">
+          <UplinkItem i={cats[lucky_numbers[2].cat].items[lucky_numbers[2].item]}/>
+        </Flex.Item>
+        <Flex.Item px="5px" width="50%">
+          <UplinkItem i={cats[lucky_numbers[3].cat].items[lucky_numbers[3].item]}/>
+        </Flex.Item>
```

Comment:
TODO for myself, I should make these not hard mapped in... These should be done based on the length of lucky_numbers

Reply:
Well, thats as close as I'm gonna get it